### PR TITLE
Fix unreachable branch in api/v1/owners/gem endpoint when owner is not found

### PR DESCRIPTION
**Pull request by esparta**
The gem method is currently using User.find_by_slug! to find the user fromthe :handle parameter. #find_by_slug will trigger a AR::NotFound exception if is not able to find by id nor handle. Since is an exception the now-removed branch is not reachable, ergo not needed.

Added some functional test to cover this case.

### CODE REVIEW

**Comment by esparta** [comment]
Reviewing the use case, if the API specs requires to return the plain, then a `find_by_slug` (no bang) should be implemented.

File: app/controllers/api/v1/owners_controller.rb [hunk 1]
```diff
@@ -37,14 +37,10 @@ def destroy
 
   def gems
     user = User.find_by_slug!(params[:handle])
-    if user
```
Comment (colby-swandale) [inline app/controllers/api/v1/owners_controller.rb:40]: As you've pointed out already, we should keep consistency and keep the plain error but change the `find_by` to not raise an exception.

Comment (esparta) [reply app/controllers/api/v1/owners_controller.rb:40]: fixed.
Added `User.find_by_slug`, adding their unit tests, then going back to the fallback using non-exceptionable method.

**Comment by sxksa6x** [comment]
Hello

**Comment by colby-swandale** [comment]
Sorry for the delay. Thanks for the contribution!

**Comment by esparta** [comment]
No problem, thanks for your help to make this happening, @colby-swandale !

**Pull request merged**